src/plugins/application_config/server/opensearch_config_client.test.ts

```diff
@@ -0,0 +1,202 @@
+
+      const value = await client.getConfig();
+
+      expect(JSON.stringify(value)).toBe(JSON.stringify({ config1: 'value1', config2: 'value2' }));
+    });
+
+    it('throws error when opensearch errors happen', async () => {
+      const error = new Error(ERROR_MESSAGE);
+
+      opensearchClient.asInternalUser.search.mockImplementation(() => {
+        return opensearchClientMock.createErrorTransportRequestPromise(error);
+      });
+
+      const client = new OpenSearchConfigurationClient(opensearchClient, INDEX_NAME, logger);
+
+      await expect(client.getConfig()).rejects.toThrowError(ERROR_MESSAGE);
+    });
+  });
+
+  describe('getEntityConfig', () => {
+    it('return configuration value from the document in the index', async () => {
+      opensearchClient.asInternalUser.get.mockImplementation(() => {
+        return opensearchClientMock.createSuccessTransportRequestPromise({
+          _source: {
+            value: 'value1',
+          },
+        });
+      });
+
+      const client = new OpenSearchConfigurationClient(opensearchClient, INDEX_NAME, logger);
+
+      const value = await client.getEntityConfig('config1');
+
+      expect(value).toBe('value1');
+    });
+
+    it('throws error when opensearch errors happen', async () => {
+      const error = new Error(ERROR_MESSAGE);
+
+      opensearchClient.asInternalUser.get.mockImplementation(() => {
+        return opensearchClientMock.createErrorTransportRequestPromise(error);
+      });
+
+      const client = new OpenSearchConfigurationClient(opensearchClient, INDEX_NAME, logger);
+
+      await expect(client.getEntityConfig('config1')).rejects.toThrowError(ERROR_MESSAGE);
+    });
+  });
+
+  describe('deleteEntityConfig', () => {
+    it('return deleted entity when opensearch deletes successfully', async () => {
+      opensearchClient.asCurrentUser.delete.mockImplementation(() => {
+        return opensearchClientMock.createSuccessTransportRequestPromise({});
+      });
+
+      const client = new OpenSearchConfigurationClient(opensearchClient, INDEX_NAME, logger);
+
+      const value = await client.deleteEntityConfig('config1');
+
+      expect(value).toBe('config1');
+    });
+
+    it('return deleted document entity when deletion fails due to index not found', async () => {
+      const error = new ResponseError({
+        statusCode: 401,
+        body: {
+          error: {
+            type: 'index_not_found_exception',
+          },
+        },
+        warnings: [],
+        headers: {
+          'WWW-Authenticate': 'content',
+        },
+        meta: {} as any,
+      });
+
+      opensearchClient.asCurrentUser.delete.mockImplementation(() => {
+        return opensearchClientMock.createErrorTransportRequestPromise(error);
+      });
+
+      const client = new OpenSearchConfigurationClient(opensearchClient, INDEX_NAME, logger);
+
+      const value = await client.deleteEntityConfig('config1');
+
+      expect(value).toBe('config1');
```

Comment:
why it is returning entity when delete fails? Doesn't it throw error or reason to fail in response?

I see in test case just above this test case, return value from `deleteEntityConfig` is same. Then how would we differentiate between successful and failed delete operation?

Reply:
When the client throws error, we perform two additional checks. The first check is whether the error is due to `index` not found. The second check is whether the error is due to `document` not found. In both cases, since the system already has no those resources to delete, we treat it as successful. 

If the error is not caused by two cases above, we will throw the error.